Htsget chart (#1)
NBISweden/htsget-rs-helm: README.md

```sh
kubectl apply -f .github/integration/scripts/charts/dependencies.yaml
```

To test TLS features for the ingress, you can use the following command to install the charts:
```sh
helm install htsget charts/htsget-rs/ --set ingress.clusterIssuer=cert-issuer
```
After the pod is Ready, you should be able to run the following command
```sh
curl -s -k -v https://htsget.local/reads/service-info
```
Since this test uses a self-signed certificate, you will need to use the `-k` flag to ignore the certificate verification.

To test TLS features for the htsget-rs service:
First comment out the `tls` block of the `ingress` block in the `values.yaml` file. Then you can use the following command to install the charts:
```sh
helm install htsget charts/htsget-rs/ --set tls.clusterIssuer=cert-issuer --set tls.enabled=true --set htsget.tls.ticketServer.key="tls.key" --set htsget.tls.ticketServer.cert="tls.crt"
```
After the pod is Ready, you should be able to run the following command
```sh
kubectl run -ti curl --image=curlimages/curl -- sh
```
and within this pod, run the following command
```sh
curl -k https://htsget-htsget-rs:8080/reads/service-info
```

> to get the htsget response.

Reviewer: ```suggestion
```

This needs to be replaced with detailed information about the values file and what parts are required and used where.
[like this](https://github.com/grafana/helm-charts/tree/main/charts/grafana#readme)

Author: Indeed, now done. I kept the original README contents but placed them in a different file in case someone finds them useful.

## Development environment

This development environment has been tested in minikube. Instructions for
installing minikube can be found in the
[official instructions](https://minikube.sigs.k8s.io/docs/start/).

### Setting up the environment and starting the minikube cluster on Ubuntu

1. Start the minikube and we use [calico](https://docs.tigera.io/calico/latest/getting-started/kubernetes/minikube) for the networkpolicy

```
minikube start --network-plugin=cni --cni=calico
```
2. Enable the addons
```
minikube addons enable ingress
minikube addons enable ingress-dns
```
3. Add the `minikube-ip` as a DNS server
```
echo -e $"\nsearch local\nnameserver $(minikube ip)\ntimeout 5" \
| sudo tee -a /etc/resolv.conf
```
4. Add minikibe 'htsget domain name ' to host file
```
echo -e  $"\n$(minikube ip) htsget.local\n$(minikube ip)" | sudo tee -a /etc/hosts
```
When you are done with testing, you can cleanup `resolv.conf` or wait till this is done automatically when the dhcp lease is renewed. The file `/etc/hosts` needs to be cleaned up manually.

### Installing the htsget-rs helm chart

> Run `helm install charts/htsget-rs` to install the system with the `htsget-rs` prefix.

Reviewer 2: ```suggestion
Run `helm install htsget-rs charts/htsget-rs` to install the  `htsget-rs` chart.
```

You can use the `kubectl get pods` command to see the kubernetes pods come online. After that, you can verify the deployment with `curl -s http://htsget.local/reads/service-info`.

### Testing with sda-download

> Start by checking out [this branch}(https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/feat/reencrypt-header) of the GDI starter-kit-storage&interfaces repository. This branch contains the re-encryption of file headers in the sda-download service. Then run:

Reviewer 2: ```suggestion
Start by checking out [this branch}(https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/featute/test-htsget-reencrypt) of the GDI starter-kit-storage&interfaces repository. This branch contains the re-encryption of file headers in the sda-download service.
Follow the instructions in the README of that repository to set-up all necessary configuration files. Then run:
```

```sh
docker compose -f docker-compose-demo.yml up
```

> and wait until the `data_loader` comletes with `exit 0`.

Reviewer 2: ```suggestion
and wait until the `data_loader` completes with `exit 0`.
```

> Then paste the following into charts/htsget-rs/configmap.yaml:

Reviewer 2: ```suggestion
Then paste the following into charts/htsget-rs/templates/configmap.yaml:
```

> Then paste the following into charts/htsget-rs/configmap.yaml:

Reviewer 2: already there?
